lua/autorun/server/toggle_voice_init.lua

```diff
@@ -49,16 +45,58 @@ hook.Add( "PlayerCanHearPlayersVoice", "CFC_ToggleLocalVoice_CanHear", function(
     local shouldUseLocal = forceLocalVoice or playerConfig[listener] or playerConfig[speaker] or playerConfigOverride[listener] or playerConfigOverride[speaker]
     if not shouldUseLocal then return end
 
-    return canHear( listener, speaker ), config.VOICE_3D
+    return canHear( listener, speaker ), playerConfig.hear3D
 end, HOOK_LOW )
 
-hook.Add( "PlayerDisconnected", "CFC_ProximityVoice_CleanupTables", function(ply)
+hook.Add( "PlayerDisconnected", "CFC_ProximityVoice_CleanupTables", function( ply )
     playerConfig[ply] = nil
     playerConfigOverride[ply] = nil
 end )
 
+util.AddNetworkString( "proximity_voice_settings_changed" )
+net.Receive( "proximity_voice_settings_changed", function( len, ply )
+    local enabled = net.ReadBool()
+    local hearDead = net.ReadBool()
+    local hear3D = net.ReadBool()
+    local range = net.ReadInt( 13 )
+
+    playerConfig[ply] = playerConfig[ply] or {}
+    playerConfig[ply] = {
+        enabled = enabled or nil,
+        hearDead = hearDead or nil,
+        hear3D = hear3D or nil,
```

Comment:
doing or nil is pointless

Reply:
was just expanding on what was already there, useless now. Thanks for the spot